internal/pkg/config/env_defaults.go

```diff
@@ -185,6 +185,9 @@ func loadLimits(agentLimit int) *envLimits {
 }
 
 func loadLimitsForAgents(agentLimit int) *envLimits {
+	if agentLimit == 0 {
+		return defaultEnvLimits()
+	}
```

Comment:
This change resolves a nitpick I have with the startup process.

Without the change, the fleet-server proc outputs:
```
{"level":"info","time":"2023-01-05T10:30:28-08:00","message":"No applicable limit for 0 agents, using default."}
{"level":"info","time":"2023-01-05T10:30:28-08:00","message":"No applicable limit for 0 agents, using default."}
{"level":"info","time":"2023-01-05T10:30:28-08:00","message":"No applicable limit for 0 agents, using default."}
{"level":"info","time":"2023-01-05T10:30:28-08:00","message":"No applicable limit for 0 agents, using default."}
10:30:28.650 INF Boot fleet-server args=["-c","fleet-server.yml"] commit=603c9df ecs.version=1.6.0 exe=./build/binaries/fleet-server-8.7.0-SNAPSHOT-darwin-x86_64/fleet-server pid=77997 ppid=69589 service.name=fleet-server version=8.7.0-SNAPSHOT
```

Where the first 4 lines are output when the go-ufcg and config.Config structs are merging as go-ufcg calls `InitDefaults` on any structs it unpacks, and may of these calls contain a call to `loadLimits(0)` which would result in the log call on line 202 executing before the logger is setup.
Adding the if statement silences log output when it's called with a 0 so that the fleet-server is able to setup its logger with the config before anything is written